### ISSUE TITLE
[Code Correction] Important changes made to split generation in VIPeR

### ISSUE THREAD

**Issue by Giounona** [KaiyangZhou/deep-person-reid#65]
in torchreid/data_manager/viper.py line 152

split = {'train': train, 'query': test, 'gallery': test,
                         'num_train_pids': num_train_pids,
                         'num_query_pids': num_pids - num_train_pids,
                         'num_gallery_pids': num_pids - num_train_pids
                         }

should it not be split = {'train': train, 'query': test[0::2], 'gallery': test[1::2]   ?
so that we have only one camera view in the query and the other camera in gallery
otherwise the recognition rate at rank 1 is lower than expected.

**Comment by KaiyangZhou** [comment]
You are right. I have corrected the code.

The code still generates 10 random splits in terms of train set and test set. The following pseudo code shows how each split is generated for viper:
```
1. for split = 0 to 9 do
2.    randomly split 632 ids to 316 ids (632 imgs) for training and 316 ids (632 imgs) for test.
3.    for the 316 test ids do:
4.        create 1st split using cameraA as query and cameraB as gallery.
5.        create 2nd split using cameraB as query and cameraA as gallery.
6.    end for
7. end for
```

As a result, there will be totally 20 splits with split_id=0~19.

You can train models on split_id=[0, 2, 4, 6, 8, 10, 12, 14, 16, 18] as consecutive two splits share the same training set.  For evaluation, simply apply models trained on split_id=0 to split_id=0&1, and models trained on split_id=2 to split_id=2&3, and so on and so forth. The final results are averaged over the 20 splits. Doing so can fairly compare your models with published methods.

Hope this is clear.
